Medium/question_1.js

```diff
@@ -0,0 +1,32 @@
+// Polymorphism is a key concept in object-oriented programming that allows objects of different types to be
+// treated as objects of a common base type. In statically-typed languages like TypeScript, polymorphism is
+// enforced through type inheritance and method signatures. In JavaScript, which is dynamically typed, you can
+// achieve polymorphism through duck typing, where objects are treated based on their behavior rather than their
+// types.
+
+class Bird {
+  fly() {
+    console.log("A bird can fly.");
+  }
+}
+
+class Dog {
+  bark() {
+    console.log("A dog can bark.");
+  }
+}
+
+function makeNoise(animal) {
+  // The makeNoise function expects an object with a "makeNoise" method
+  if (animal && typeof animal.makeNoise === "function") {
+    animal.makeNoise();
```

Comment:
`if (animal && typeof animal.bark === "function") animal.bark();`

Reply:
okay, let me correct this one, and thank you for the suggestions I'll work on them for sure.

Reply:
After you made changes, mark each comment resolved and request review again.
<img width="350" alt="Screenshot 2023-11-09 at 8 58 28 AM" src="https://github.com/saadali18/OOP-interview-questions/assets/61346412/73d82fb7-a4ef-4378-9131-247bdfb45a72">